db/sqlinterfaces.c

```diff
@@ -7755,16 +7755,36 @@ static int process_set_commands(struct sqlclntstate *clnt)
                     rc = ii + 1;
                 }
             } else if (strncasecmp(sqlstr, "user", 4) == 0) {
+                char *savptr = sqlstr;
                 sqlstr += 4;
                 sqlstr = cdb2_skipws(sqlstr);
-                clnt->have_user = 1;
-                strncpy(clnt->user, sqlstr, sizeof(clnt->user));
+                if (strlen(sqlstr) >= sizeof(clnt->user)) {
+                    sqlstr = savptr;
+                    logmsg(LOGMSG_USER,
```

Comment:
I’d expect a test that fails with this error.

Reply:
This is printed on the .db log file. It would be nice to return that to user.

Reply:
Changing errstr to return that.